xdist/newhooks.py

```diff
@@ -44,3 +45,9 @@ def pytest_testnodedown(node, error):
 def pytest_xdist_node_collection_finished(node, ids):
     """called by the master node when a node finishes collecting.
     """
+
+
+@pytest.mark.firstresult
+def pytest_xdist_make_scheduler(numnodes, log, config):
```

Comment:
I think we can drop `numnodes` from the hook signature... it can be easily obtained from the `config` object.